src/utils/byte_parsing.ts

```diff
@@ -320,33 +320,33 @@ function itole4(num : number) : Uint8Array {
 }
 
 /**
- * @param {string} uuid
- * @returns {string}
- * @throws AssertionError - The uuid length is not 16
+ * Convert little-endian GUID into big-endian UUID.
+ * @param {Uint8Array} guid
+ * @returns {Uint8Array} - uuid
+ * @throws AssertionError - The guid length is not 16
  */
-function guidToUuid(uuid : string) : string {
-  assert(uuid.length === 16, "UUID length should be 16");
-  const buf = strToBytes(uuid);
+function guidToUuid(guid : Uint8Array) : Uint8Array {
+  assert(guid.length === 16, "GUID length should be 16");
 
-  const p1A = buf[0];
-  const p1B = buf[1];
-  const p1C = buf[2];
-  const p1D = buf[3];
-  const p2A = buf[4];
-  const p2B = buf[5];
-  const p3A = buf[6];
-  const p3B = buf[7];
-  const p4 = buf.subarray(8, 10);
-  const p5 = buf.subarray(10, 16);
+  const p1A = guid[0];
+  const p1B = guid[1];
+  const p1C = guid[2];
+  const p1D = guid[3];
+  const p2A = guid[4];
+  const p2B = guid[5];
+  const p3A = guid[6];
+  const p3B = guid[7];
+  const p4 = guid.subarray(8, 10);
+  const p5 = guid.subarray(10, 16);
 
-  const ord = new Uint8Array(16);
-  ord[0] = p1D; ord[1] = p1C; ord[2] = p1B; ord[3] = p1A; // swap32 BE -> LE
-  ord[4] = p2B; ord[5] = p2A;                             // swap16 BE -> LE
-  ord[6] = p3B; ord[7] = p3A;                             // swap16 BE -> LE
-  ord.set(p4,  8);
-  ord.set(p5, 10);
+  const uuid = new Uint8Array(16);
+  uuid[0] = p1D; uuid[1] = p1C; uuid[2] = p1B; uuid[3] = p1A;
+  uuid[4] = p2B; uuid[5] = p2A;
+  uuid[6] = p3B; uuid[7] = p3A;
```

Comment:
The old comments added some information about the strange of behavior of reversing the order by-parts (even if the variable names also give a cue).

Other than that, doing that operation on a byte array do seems more portable and logical

Reply:
It seems that the older comment where wrong (i think that the convertion is LE => BE and not BE => LE), so i just removed them. Maybe i could write a little comment to replace them.